daemons/controld/controld_schedulerd.c

```diff
@@ -494,11 +494,11 @@ do_pe_invoke_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void
         crm_xml_add_int(output, PCMK_XA_NO_QUORUM_PANIC, 1);
     }
 
-    rc = pcmk_rc2legacy(pcmk_schedulerd_api_graph(schedulerd_api, output, &ref));
-
-    if (rc < 0) {
+    rc = pcmk_schedulerd_api_graph(schedulerd_api, output, &ref);
+    if (rc != pcmk_rc_ok) {
+        free(ref);
```

Comment:
It's possible for pcmk_schedulerd_api_graph to return an error before ref has been allocated - for instance, if it returns ENOTCONN.

Reply:
yep, but free(NULL) is fine